# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/TracingAPI.h
# Commit: 662cb63c42d7
# Full Hash: 662cb63c42d7874a79256a6edf1a03d83e43a014
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 2
# Description:
#   Bug 1692065 - wasm: Create a unique wasm::AnyRef type. r=yury,jonco
#   
#   Right now wasm references area represented as a JSObject*. This is assumed in
#   many places in SpiderMonkey, most prominently in our GC integration
#   around tracing, the store buffer, and write barriers. This will cause issues
# ==============================================================================

diff -r 8b506ab41451 -r 662cb63c42d7 js/public/TracingAPI.h
--- a/js/public/TracingAPI.h	Fri Aug 04 13:54:57 2023 +0000
+++ b/js/public/TracingAPI.h	Fri Aug 04 14:06:49 2023 +0000
@@ -308,6 +308,9 @@
 
 class AbstractGeneratorObject;
 class SavedFrame;
+namespace wasm {
+class AnyRef;
+}  // namespace wasm
 
 namespace gc {
 
@@ -378,6 +381,7 @@
 // to not be *actual* overloads, but for the moment we still declare them here.
 JS_DECLARE_TRACE_ROOT(js::AbstractGeneratorObject*)
 JS_DECLARE_TRACE_ROOT(js::SavedFrame*)
+JS_DECLARE_TRACE_ROOT(js::wasm::AnyRef)
 
 #undef JS_DECLARE_TRACE_ROOT
 